src/network/networkd-sysctl.c

```diff
@@ -167,6 +167,18 @@ static int link_set_ipv6_hop_limit(Link *link) {
         return sysctl_write_ip_property_int(AF_INET6, link->ifname, "hop_limit", link->network->ipv6_hop_limit);
 }
 
+static int link_set_ipv6_retrans_time(Link *link) {
+        assert(link);
+
+        if (!link_is_configured_for_family(link, AF_INET6))
+                return 0;
+
+        if (link->network->ipv6_retrans_time <= 0)
+                return 0;
+
+        return sysctl_write_ip_neigh_property_int(AF_INET6, link->ifname, "retrans_time_ms", link->network->ipv6_retrans_time);
```

Comment:
hmm, the ipv6_retrans_time field is "uint32_t", i.e. unsigned. But here you silently cast it to an `int`, i.e. signed. This means we'll end up with negative values here possibly...

Reply:
While sysctl can handle the negative numbers, I've reworked this to be explicit about this value being unsigned.